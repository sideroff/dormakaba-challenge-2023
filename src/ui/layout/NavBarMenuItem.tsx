import { useRouter } from 'next/router';
import Link from 'next/link';
import { MouseEventHandler } from 'react';
import MenuItem from '@mui/material/MenuItem';
import Typography from '@mui/material/Typography';

interface NavBarMenuItemProps {
  label: string;
  href: string;
  onClick: MouseEventHandler | undefined;
}

export function NavBarMenuItem({ label, href, onClick }: NavBarMenuItemProps) {
  const { pathname } = useRouter();

  return (
    <MenuItem
      selected={pathname === href}
      onClick={onClick}
      component={Link}
      href={href}
    >
      <Typography textAlign="center">{label}</Typography>
    </MenuItem>
  );
}
