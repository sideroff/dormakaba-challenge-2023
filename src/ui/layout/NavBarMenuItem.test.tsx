import { render } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import { NavBarMenuItem } from './NavBarMenuItem';

jest.mock('next/router', () => require('next-router-mock'));

describe('NavBarMenuItem', () => {
  it('should render correctly', () => {
    mockRouter.push('/test');

    const onNavBarMenuItemClick = jest.fn();

    const { container } = render(
      <NavBarMenuItem
        label="Test"
        href="/test"
        onClick={onNavBarMenuItemClick}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
