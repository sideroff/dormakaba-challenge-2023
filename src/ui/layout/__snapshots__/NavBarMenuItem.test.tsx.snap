// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarMenuItem should render correctly 1`] = `
<a
  class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters Mui-selected MuiMenuItem-root MuiMenuItem-gutters Mui-selected css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
  href="/test"
  role="menuitem"
  tabindex="-1"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
  >
    Test
  </p>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</a>
`;
